src/selectors.js

```diff
@@ -74,29 +94,84 @@ const _buildRecords = (ents, key, opts) => {
 
 const defaultArr = [];
 
-const buildRecords = (ents = {}, key, opts = {}) => {
-  // console.log(`All. key: ${key}`, ents, opts)
-  if (!_.isObject(opts)) opts = {};
-  const { model, defaultJoins } = getModel(key);
-  const joinKeys = utils.composeArray(opts.joins, defaultJoins);
-  let final = enrichRecords(ents, key) || defaultArr;
-
-  joinKeys.forEach((k) => {
-    const mergeItems = enrichRecords(ents, k);
-    final = joins({
-      items: final,
-      mergeItems,
+const buildRecords = (entities = {}, key, opts = {}) => {
+  opts = _.isObject(opts) ? opts : {};
+  const { defaultJoins = [] } = getModel(key) || {};
+  let finalRecords = enrichRecords(entities, key) || defaultArr;
+  const joinsArray = utils.composeArray(defaultJoins, opts.joins);
+  const joinKeys = buildNestedStructure(joinsArray, key);
+
+  joinKeys.forEach(({ key: joinKey, children }) => {
+    const enrichedJoins = children.reduce((acc, child) => {
+      const parentRecords = acc.length
+        ? acc
+        : enrichRecords(entities, child.parent);
+      const childRecords = enrichRecords(entities, child.key);
+      return joins({
+        items: parentRecords,
+        mergeItems: childRecords,
+        tableName: child.parent,
+        mergeTableName: child.key,
+      });
+    }, []);
+
+    const fullRecords = enrichedJoins.map((record) =>
+      buildFullRecord(
+        record,
+        joinKey,
+        children.map((c) => c.key)
+      )
+    );
+
+    finalRecords = joins({
+      items: finalRecords,
+      mergeItems: fullRecords.length
+        ? fullRecords
```

Comment:
dit is eigenlijk het onderscheid tussen joins/ joins through toch? fullRecords is leeg als het een simpele joins is, of zie ik dat verkeerd? Zou nice zijn om duidelijker te maken in de flow waarom deze conditie erin zit (bv door namen te geven)

Reply:
ja klopt inderdaad. Zal het meenemen